pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -575,6 +589,12 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 				// The binding is not completed; let PVC sync handle it
 				glog.V(4).Infof("synchronizing PersistentVolume[%s]: volume not bound yet, waiting for syncClaim to fix it", volume.Name)
 			} else {
+				if isMisMatch, err := checkVolumeModeMisMatches(&claim.Spec, &volume.Spec); err != nil || isMisMatch {
+					msg := fmt.Sprintf("Dynamically provisioned PersistentVolume[%s] has an incompatible volumeMode to PersistentVolumeClaim[%s]. It won't be bound to the claim. Please delete this volume manually.", volume.Name, claim.Name)
```

Comment:
I think we can just have an event on the PVC, and not on the PV object.  We have the claim object here, so I don't think we need the check in `syncUnboundClaim`

How about a message like:
"Dynamically provisioned volume[%s] has incompatible volumeMode. This provisioner may not support the requested volumeMode."

Reply:
>I think we can just have an event on the PVC, and not on the PV object. We have the claim object here, so I don't think we need the check in syncUnboundClaim

One of the aims for my adding check code in syncUnboundClaim was to skip calling provisionClaim repeatedly
after one volume has already been provisioned.

After I read your comment, I checked the provisionClaim codes for both internal and external provisioner again,
and found that both of them are implemented to provision a volume named "pvc-" + string(claim.UID),
and they won't try to create another one if a volume with the name is found. 
In addition, it exists in provisoner independent part of the code, or in provisionClaimOperation().
(For internal provisioner: pkg/controller/volume/persistentvolume/pv_controller.go, and 
 for external provisioner: lib/controller/controller.go )

Therefore, it looks safe enough to call provisionClaim without any checks here, because enough check is done in provisionClaimOperation,
so I agree to putting it only in syncVolume side.

>How about a message like:
>"Dynamically provisioned volume[%s] has incompatible volumeMode. This provisioner may not support the requested volumeMode."

Changed as suggested.

Please review v5 patch below.